frontend/src/pages/ProfilePage.js

```diff
@@ -9,45 +11,75 @@ function ProfilePage() {
   const [passModalOpen, setPassModalOpen] = useState(false);
   const [deleteModalOpen, setDeleteModalOpen] = useState(false);
 
-  const user = {
-    full_name: "Carly Shay",
-    email: "carlyshay@gmail.com",
-    join_date: "June 2019",
-    profilePicture: "",
-  };
+  const [user, setUser] = useState({});
+  const [isCurrentUser, setIsCurrentUser] = useState(false);
+  // const [isCurrentUserAdmin, setIsCurrentUserAdmin] = useState(true); // change this once andrew's pr gets merged in
+
+  const [active, setActive] = useState(history.location.pathname.split("/profile/")[1]);
+
+  const navigate = useNavigate();
+
+  async function getUser(id) {
+    const res = await api_user(id);
+    if (res === null) navigate("/user-not-found");
+    return res;
+  }
+
+  useEffect(async () => {
+    const res = await getUser(active);
+    setUser(res.user);
```

Comment:
In case the request fails, we should validate that a valid response was received

```suggestion
    if(!res || !res.user) return;
    setUser(res.user);
```

Reply:
Actually, to make this even cleaner you could remove the `getUser` function and just do:

```js
useEffect(async () => {
    const res = await api_user(id);
    if(!res || !res.user) navigate("/user-not-found");
    else {
        setUser(res.user);
        setIsCurrentUser(res.sameUser);
    }
}, [active]);